icon_validator/rules/support_validator.py

```diff
@@ -0,0 +1,39 @@
+from .validator import KomandPluginValidator
+
+
+class SupportValidator(KomandPluginValidator):
+
+    @staticmethod
+    def validate_support(support):
+        lsupport = support.lower()
+        if lsupport == 'komand':
+            raise Exception("Support 'komand' not allowed. It's likely you meant 'rapid7'.")
+        if support.endswith('.'):
+            raise Exception('Support ends with period when it should not.')
+        if not support[0].islower():
+            raise Exception('Support starts with a capital letter when it should not.')
+        if " " in support:
+            raise Exception('Support should be separated by underscores, not spaces.')
+
+    @staticmethod
+    def validate_support_quotes(spec):
+        '''Requires raw spec to see the quotes'''
+        for line in spec.splitlines():
+            if line.startswith('support:'):
+                val = line[line.find(' ') + 1:]
+                if '"' in val:
+                    raise Exception('Support is surrounded by or contains quotes when it should not')
+                if "'" in val:
+                    raise Exception('Support is surrounded by or contains quotes when it should not')
```

Comment:
These two cases can be combined:
```
if '"' in val or "'" in val:
    raise blah
```
Interesting as well, you could do a set comparison to check:
```
if {"'", '"'}.intersection(set(val)):
    raise blah
```

Reply:
Additionally, make sure to use punctuation consistently within your error messages. Some of them are using it and some are not.